internal/wasm/function_definition.go

```diff
@@ -0,0 +1,151 @@
+package wasm
+
+import (
+	"github.com/tetratelabs/wazero/api"
+)
+
+// ImportedFunctions returns the definitions of each imported function.
+func (m *Module) ImportedFunctions() (ret []api.FunctionDefinition) {
+	for _, d := range m.functionDefinitions {
+		if d.importDesc != nil {
+			ret = append(ret, d)
+		}
+	}
+	return
+}
+
+// ExportedFunctions returns the definitions of each exported function.
+func (m *Module) ExportedFunctions() (ret []api.FunctionDefinition) {
+	for _, d := range m.functionDefinitions {
+		if d.exportNames != nil {
+			ret = append(ret, d)
+		}
+	}
+	return
+}
+
+// BuildFunctionDefinitions generates function metadata that can be parsed from
+// the module. This must be called after all validation.
+//
+// Note: This is exported for tests who don't use wazero.Runtime or
+// NewHostModule to compile the module.
+func (m *Module) BuildFunctionDefinitions() *Module {
```

Comment:
do we need to return the self pointer?

Reply:
no, just did it for convenience when inlining wasm modules